schema_based_on_diagram.sql

```diff
@@ -0,0 +1,69 @@
+CREATE TABLE "patients"(
+    "id" INTEGER NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "date_of_birth" DATE NOT NULL
+);
+ALTER TABLE
+    "patients" ADD PRIMARY KEY("id");
+CREATE TABLE "invoices"(
+    "id" INTEGER NOT NULL,
+    "total_amount" DECIMAL(8, 2) NOT NULL,
+    "generated_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "payed_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "medical_history_id" INTEGER NOT NULL
+);
+ALTER TABLE
+    "invoices" ADD PRIMARY KEY("id");
+CREATE TABLE "medical_histories"(
+    "id" INTEGER NOT NULL,
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER NOT NULL,
+    "status" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "medical_histories" ADD PRIMARY KEY("id");
+CREATE TABLE "treatments"(
+    "id" INTEGER NOT NULL,
+    "type" VARCHAR(255) NOT NULL,
+    "name" VARCHAR(255) NOT NULL
+);
+ALTER TABLE
+    "treatments" ADD PRIMARY KEY("id");
+CREATE TABLE medical_history_treatments (
+  medical_history_id INT,
+  treatment_id INT,
+  PRIMARY KEY (medical_history_id, treatment_id),
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments (id)
+);
+CREATE TABLE "invoice_items"(
+    "id" INTEGER NOT NULL,
+    "unit_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL,
+    "treatment_id" INTEGER NOT NULL
+);
+ALTER TABLE
+    "invoice_items" ADD PRIMARY KEY("id");
+ALTER TABLE
+    "medical_histories" ADD CONSTRAINT "medical_histories_patient_id_foreign" FOREIGN KEY("patient_id") REFERENCES "patients"("id");
+ALTER TABLE
+    "invoices" ADD CONSTRAINT "invoices_medical_history_id_foreign" FOREIGN KEY("medical_history_id") REFERENCES "medical_histories"("id");
+ALTER TABLE
+    "invoice_items" ADD CONSTRAINT "invoice_items_invoice_id_foreign" FOREIGN KEY("invoice_id") REFERENCES "invoices"("id");
+ALTER TABLE
+    "invoice_items" ADD CONSTRAINT "invoice_items_treatment_id_foreign" FOREIGN KEY("treatment_id") REFERENCES "treatments"("id");
+ALTER TABLE
+    "treatments" ADD CONSTRAINT "treatments_id_foreign" FOREIGN KEY("id") REFERENCES "medical_histories"("id");
+CREATE INDEX idx_medhist_id ON medical_history_treatments (medical_history_id);
+
+CREATE INDEX idx_treatment_id ON medical_history_treatments (treatment_id);
+
+CREATE INDEX idx_invoice_items_invoice_id ON invoice_items (invoice_id);
+
+CREATE INDEX idx_invoice_items_treatment_id ON invoice_items (treatment_id);
+
+CREATE INDEX idx_invoices_medicalhistory_id ON invoices (medical_history_id);
+
+CREATE INDEX idx_patient_id ON medical_histories (patient_id);
```

Comment:
- [ ] [ OPTIONAL] For more automation and ease of scaling, you may not have to include the `index value/name` so that it gets automatically generated. E.g: `CREATE INDEX ON medical_histories (patient_id)`, etc.